examples/sv3d/models/instantmesh/models/decoder/transformer.py

```diff
@@ -0,0 +1,136 @@
+            dropout=attn_drop,
+            has_bias=attn_bias,
+            batch_first=True,
+        )
+        self.norm2 = nn.LayerNorm([inner_dim])
+        self.self_attn = nn.MultiheadAttention(
+            embed_dim=inner_dim, num_heads=num_heads, dropout=attn_drop, has_bias=attn_bias, batch_first=True
+        )
+        self.norm3 = nn.LayerNorm([inner_dim])
+        self.mlp = nn.SequentialCell(
+            nn.Dense(inner_dim, int(inner_dim * mlp_ratio)),
+            nn.GELU(),
+            nn.Dropout(p=mlp_drop),
+            nn.Dense(int(inner_dim * mlp_ratio), inner_dim),
+            nn.Dropout(p=mlp_drop),
+        )
+
+    def construct(self, x, cond):
+        # x: [N, L, D]
+        # cond: [N, L_cond, D_cond]
+        x = x + self.cross_attn(self.norm1(x), cond, cond)[0]
+        before_sa = self.norm2(x)
+        x = x + self.self_attn(before_sa, before_sa, before_sa)[0]
+        x = x + self.mlp(self.norm3(x))
+        return x
+
+
+class TriplaneTransformer(nn.Cell):
+    """
+    Transformer with condition that generates a triplane representation.
+
+    Reference:
+    Timm: https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/vision_transformer.py#L486
+    """
+
+    def __init__(
+        self,
+        inner_dim: int,
+        image_feat_dim: int,
+        triplane_low_res: int,
+        triplane_high_res: int,
+        triplane_dim: int,
+        num_layers: int,
+        num_heads: int,
+        eps: float = 1e-6,
+    ):
+        super().__init__()
+
+        # attributes
+        self.triplane_low_res = triplane_low_res
+        self.triplane_high_res = triplane_high_res
+        self.triplane_dim = triplane_dim
+
+        # modules
+        # initialize pos_embed with 1/sqrt(dim) * N(0, 1)
+        self.pos_embed = ms.Parameter(ops.randn(1, 3 * triplane_low_res**2, inner_dim) * (1.0 / inner_dim) ** 0.5)
+        self.layers = nn.CellList(
+            [
+                BasicTransformerBlock(inner_dim=inner_dim, cond_dim=image_feat_dim, num_heads=num_heads, eps=eps)
+                for _ in range(num_layers)
+            ]
+        )
+        self.norm = nn.LayerNorm([inner_dim], epsilon=eps)
+        self.deconv = nn.Conv2dTranspose(
+            inner_dim, triplane_dim, kernel_size=2, stride=2, padding=0, pad_mode="valid", has_bias=True
+        )
+
+    def construct(self, image_feats):
+        # image_feats: [N, L_cond, D_cond]
+
+        N = image_feats.shape[0]
+        H = W = self.triplane_low_res
+        # L = 3 * H * W
+
+        x = self.pos_embed.tile((N, 1, 1))  # [N, L, D]
+        for layer in self.layers:
+            x = layer(x, image_feats)
+        x = self.norm(x)
+
+        # separate each plane and apply deconv
+        x = x.view(N, 3, H, W, -1)
+        # x = ops.einsum('nihwd->indhw', x)  # [3, N, D, H, W]
+        x = ops.reshape(x, (3, N, -1, H, W))
+        x = x.view(3 * N, -1, H, W)  # [3*N, D, H, W]
+        x = self.deconv(x)  # [3*N, D', H', W']
```

Comment:
Currently, I found that a bug was in this line. The comparison of the inputs before/after this line is shown below.
| x.sum() | torch (.87) | ms (.214) |
| ------- | ----------- | :--------: |
| before  | -4436.81250 | -4389.7744 |
| after   | 91614.03125 |  835.584  |

I checked that the operator `nn.Conv2dTranspose` works correctly in behavior. I wonder was it caused by the error accumulations of each layer and eventually fell into such a huge difference after this deconv operator. Any help will be appreciated thanks.

Reply:
Solved by replacing into `ops.permute()`, my bad.